applications/HDF5Application/python_scripts/core/operations/model_part.py

```diff
@@ -105,6 +105,69 @@ def __call__(self, model_part, hdf5_file):
         KratosHDF5.HDF5ElementDataValueIO(
             self.GetSettings(model_part).Get(), hdf5_file).ReadElementResults(model_part.Elements)
 
+class ElementFlagValueOutput(VariableIO):
+    '''Writes non-historical element flag values to a file.'''
+
+    def __init__(self, settings):
+        super(ElementFlagValueOutput, self).__init__(settings)
+
+    def __call__(self, model_part, hdf5_file):
+        KratosHDF5.HDF5ElementFlagValueIO(
+            self.GetSettings(model_part).Get(), hdf5_file).WriteElementFlags(model_part.Elements)
+
+
+class ElementFlagValueInput(VariableIO):
+    '''Reads non-historical element flag values from a file.'''
+
+    def __init__(self, settings):
+        super(ElementFlagValueInput, self).__init__(settings)
+
+    def __call__(self, model_part, hdf5_file):
+        KratosHDF5.HDF5ElementFlagValueIO(
+            self.GetSettings(model_part).Get(), hdf5_file).ReadElementFlags(model_part.Elements)
+
+class ConditionDataValueOutput(VariableIO):
+    '''Writes non-historical element data values to a file.'''
+
+    def __init__(self, settings):
+        super(ConditionDataValueOutput, self).__init__(settings)
+
+    def __call__(self, model_part, hdf5_file):
+        KratosHDF5.HDF5ConditionDataValueIO(
+            self.GetSettings(model_part).Get(), hdf5_file).WriteConditionResults(model_part.Conditions)
+
+
+class ConditionDataValueInput(VariableIO):
+    '''Reads non-historical element data values from a file.'''
+
+    def __init__(self, settings):
+        super(ConditionDataValueInput, self).__init__(settings)
+
+    def __call__(self, model_part, hdf5_file):
+        KratosHDF5.HDF5ConditionDataValueIO(
+            self.GetSettings(model_part).Get(), hdf5_file).ReadConditionResults(model_part.Conditions)
+
+class ConditionFlagValueOutput(VariableIO):
+    '''Writes non-historical element flag values to a file.'''
+
+    def __init__(self, settings):
+        super(ConditionFlagValueOutput, self).__init__(settings)
+
+    def __call__(self, model_part, hdf5_file):
+        KratosHDF5.HDF5ConditionFlagValueIO(
+            self.GetSettings(model_part).Get(), hdf5_file).WriteConditionFlags(model_part.Conditions)
+
+
+class ConditionFlagValueInput(VariableIO):
+    '''Reads non-historical element flag values from a file.'''
+
+    def __init__(self, settings):
+        super(ConditionFlagValueInput, self).__init__(settings)
+
+    def __call__(self, model_part, hdf5_file):
+        KratosHDF5.HDF5ConditionFlagValueIO(
+            self.GetSettings(model_part).Get(), hdf5_file).ReadConditionFlags(model_part.Conditions)
```

Comment:
why not passing the `Communicator` here too? This could unify implementations in the `hdf5_container_component_io`
Same for other cases where you don't pass the `Communicator`

Reply:
done :)